client/src/main/java/com/battleships/start_window/connection/ServerIO.java

```diff
@@ -0,0 +1,29 @@
+package com.battleships.start_window.connection;
+
+import java.io.PrintWriter;
+import java.util.Optional;
+import java.util.Scanner;
+
+class ServerIO {
+    private PrintWriter socketWriter;
+    private Scanner socketScanner;
+
+    ServerIO(PrintWriter socketWriter, Scanner socketScanner) {
+        this.socketWriter = socketWriter;
+        this.socketScanner = socketScanner;
+    }
+
+
+    Optional<String> getMessageOptional() {
+        Optional <String> messageOptional = Optional.empty();
+        if (socketScanner.hasNextLine()) {
+            messageOptional = Optional.of(socketScanner.nextLine());
```

Comment:
You should extract result of nextLine method before using it. For a clean code and the other great stuff.